js/utils/ajax.js

```diff
@@ -19,16 +19,30 @@
  */
 
 define([
-  'jquery'
-], function($) {
+  'jquery',
+  'token-utils',
+  'user-utils'
+], function($, token, user) {
   $(document).ajaxError(function(event, jqueryXHR, error, errorThrown) {
+    var i, cluster,
+      isCurrentCluster = window.cluster && error.url.indexOf(window.cluster.api.url) > -1;
+
```

Comment:
The whole logic of this error handler is not really obvious. A large comment must be added here to explain to whole logic and all cases that should be handled and how.